apps/owmweather/interface.html

```diff
@@ -2,35 +2,34 @@
   <head>
     <link rel="stylesheet" href="../../css/spectre.min.css">
   </head>
-  <body>   
+  <body>
     <h3>Set OpenWeatherMap (OWM) API key</h3>
     <p><input id="apikey" onkeyup="checkInput()"  style="width:90%; margin: 3px"></input><button id="upload" class="btn btn-primary">Save key</button></p>
-    
+
     <h4>Where to get your personal API key?</h4>
     <p>Go to <a href="https://home.openweathermap.org/users/sign_up">https://home.openweathermap.org/users/sign_up</a> and sign up for a free account.<br>
     After registration you can login and optain your personal API key.</p>
-    
+
 
     <script src="../../core/lib/interface.js"></script>
 
     <script>
-    
+
        function checkInput() {
-        if(document.getElementById("apikey").value==="") { 
-                document.getElementById('upload').disabled = true; 
-            } else { 
+        if(document.getElementById("apikey").value==="") {
+                document.getElementById('upload').disabled = true;
+            } else {
                 document.getElementById('upload').disabled = false;
             }
         }
         checkInput();
-    
+
         var settings = {};
         function onInit(){
           console.log("Loading settings from BangleJs...");
           try {
-            Util.readStorage("owmweather.json", data=>{
-              if(data.length > 0){
-                settings = JSON.parse(data);
+            Util.readStorageJSON("owmweather.json", settings=>{
```

Comment:
I think we might have a behaviour change here to check - previously if the file didn't exist, we'd get called back with `""`, but now we'll end up trying `parseRJSON("")`.

Having said that, while `JSON.parse()` will throw on that, looking at [the code](https://github.com/espruino/EspruinoAppLoaderCore/blob/c97b7851f50cfff4e898c2264a337a17085ce463/js/utils.js#L405-L453) I think we might be ok, we only `match(EOF)` if we have a token, otherwise we just return `undefined`

Short on time atm but can give this a test later on

(same for `tinycmc`'s `interface.html` too)

Reply:
Thanks - I just tried and:

```
Utils.parseRJSON("")
=undefined
```

So I think we're fine there, *but* I really screwed this change up anyway as `settings` was a global var, which I should have updated with a `if (data) settings=data`. Hopefully I fixed that now!

Reply:
Great, on both counts there! Yeah I think we're good to go in that case